### PR TITLE
Event quality flag added to CTF; 

### PR DESCRIPTION
**Pull request by AllaMaevskaya**
Event quality flag added to CTF; 
replace assert in LookUpTable to the alarm message
removed reconstruction  debug messages

### CODE REVIEW

File: DataFormats/Detectors/FIT/FT0/include/DataFormatsFT0/Digit.h [hunk 1]
```diff
@@ -112,6 +113,7 @@ struct Digit {
   void setEventStatus(uint8_t stat) { mEventStatus = stat; };
   void setStatusFlag(EEventStatus bit, bool value) { mEventStatus |= (value << bit); };
   bool getStatusFlag(EEventStatus bit) const { return bool(mEventStatus << bit); }
+  uint8_t getEventStatusWord() const { return mEventStatus; }
   ClassDefNV(Digit, 4);
```
Comment (shahor02) [inline DataFormats/Detectors/FIT/FT0/include/DataFormatsFT0/Digit.h]: Please increase class version

File: DataFormats/Detectors/FIT/FT0/include/DataFormatsFT0/CTF.h [hunk 2]
```diff
@@ -43,12 +43,12 @@ struct CompressedDigits {
...
   std::vector<uint32_t> orbitInc; // increment in orbit
   std::vector<uint8_t> nChan;     // number of fired channels
-  // std::vector<uint8_t> eventFlags;     // special flags about event conditions: pile-up, not use for collision time, not use for event plane, etc.
+  std::vector<uint8_t> eventFlags; // special flags about event conditions: pile-up, not use for collision time, not use for event plane, etc.
 
   // channel data
   std::vector<uint8_t> idChan;   // channels ID: 1st on absolute, then increment
   std::vector<int16_t> cfdTime;  // CFD time
-  std::vector<int16_t> qtcAmpl;  // Amplitude
+  std::vector<int32_t> qtcAmpl;  // Amplitude
```
Comment (shahor02) [inline DataFormats/Detectors/FIT/FT0/include/DataFormatsFT0/CTF.h:51]: You wrote me that the amplitude is limited to +-2048, what is the reason to change it to int?

File: Detectors/FIT/FT0/reconstruction/include/FT0Reconstruction/CTFCoder.h [hunk 3]
```diff
@@ -159,24 +159,23 @@ void CTFCoder::decompress(const CompressedDigits& cd, VDIG& digitVec, VCHAN& cha
...
       auto icc = channelVec.size();
       const auto& chan = channelVec.emplace_back((chID += cd.idChan[icc]), cd.cfdTime[icc], cd.qtcAmpl[icc], cd.qtcChain[icc]);
       //
       // rebuild digit
-      if (std::abs(chan.CFDTime) < Geometry::mTime_trg_gate) {
-        if (chan.ChId < 4 * uint8_t(Geometry::NCellsA)) { // A side
-          amplA += chan.QTCAmpl;
-          timeA += chan.CFDTime;
-          trig.nChanA++;
+      //  if (std::abs(chan.CFDTime) < Geometry::mTime_trg_gate) {
```
Comment (shahor02) [inline Detectors/FIT/FT0/reconstruction/include/FT0Reconstruction/CTFCoder.h]: why?

**Review by shahor02 — COMMENTED** [review]
Hi @AllaMaevskaya 

Please see some comments/questions below.

File: DataFormats/Detectors/FIT/FT0/include/DataFormatsFT0/Digit.h (hunk 1 above)
Comment (AllaMaevskaya) [reply DataFormats/Detectors/FIT/FT0/include/DataFormatsFT0/Digit.h]: Its sum amplitude from A and C sides in TCM, 18 bits long

Comment (shahor02) [reply DataFormats/Detectors/FIT/FT0/include/DataFormatsFT0/Digit.h]: so, the range is not +-2048?

File: DataFormats/Detectors/FIT/FT0/include/DataFormatsFT0/CTF.h (hunk 2 above)
Comment (AllaMaevskaya) [reply DataFormats/Detectors/FIT/FT0/include/DataFormatsFT0/CTF.h:51]: Its sum amplitude from A and C sides in TCM, 18 bits long

File: Detectors/FIT/FT0/reconstruction/include/FT0Reconstruction/CTFCoder.h (hunk 3 above)
Comment (AllaMaevskaya) [reply Detectors/FIT/FT0/reconstruction/include/FT0Reconstruction/CTFCoder.h]: sorry, fixed now

File: DataFormats/Detectors/FIT/FT0/include/DataFormatsFT0/Digit.h (hunk 1 above)
Comment (AllaMaevskaya) [reply DataFormats/Detectors/FIT/FT0/include/DataFormatsFT0/Digit.h]: no

**Review by shahor02 — APPROVED** [review]

**Comment by shahor02** [comment]
fine for me, is it ok to merge this?

**Review by mslupeck — APPROVED** [review]
Looks OK to me too

**Pull request merged**